### ISSUE TITLE
Externals to still respect resolution

### ISSUE THREAD

**Issue by guybedford** [rollup/rollup#424]
Consider the case:

* `a.js` depends on `./b.js`
* `a.js` depends on `jquery`

`resolveId` will provide the resolutions:
* `./b.js` -> `b.js`
* `jquery` -> `npm:jquery@2.1.4/dist/jquery.js`

`externals` is set to the list `['npm:jquery@2.1.4/dist/jquery.js']`.

The expected behaviour is that since `jquery` is resolved into `npm:jquery@2.1.4/dist/jquery.js` which is provided as an external, that this module would not be loaded and would inline into the build as an import to its resolved name of `npm:jquery@2.1.4/dist/jquery.js`.

The actual behaviour is that the built file contains a reference to `jquery`, the unresolved external.

Apart from that everything else is working correctly.

The idea is to provide a way to control resolution of externals in the built file. Perhaps another mechanism may be suitable for this though.

**Comment by guybedford** [comment]
To clarify, explicitly returning `false` in `resolveId` is the only way to make this work, but I'd still like to be able to ensure that jQuery is normalized as part of the build (a valid resolve completion) so that the output gives `npm:jquery@2.1.4/dist/jquery.js` as the require.

It sounds like this is closely related to https://github.com/rollup/rollup/issues/410 pending release?

**Comment by guybedford** [comment]
Just to update here, on the latest patch, this case now fails as `resolveId` doesn't get called on the external at all. The stack is pretty much as before:

```
Error: Could not load npm:jquery@2.2.0/dist/jquery.js (imported by a.js): Cannot read property 'metadata' of undefined
    at /Users/guybedford/projects/systemjs-builder/node_modules/rollup/dist/rollup.js:7647:10
    at tryCatch (/Users/guybedford/projects/systemjs-builder/node_modules/rollup/dist/rollup.js:337:12)
    at invokeCallback (/Users/guybedford/projects/systemjs-builder/node_modules/rollup/dist/rollup.js:349:13)
    at publish (/Users/guybedford/projects/systemjs-builder/node_modules/rollup/dist/rollup.js:320:7)
    at publishRejection (/Users/guybedford/projects/systemjs-builder/node_modules/rollup/dist/rollup.js:270:3)
    at flush (/Users/guybedford/projects/systemjs-builder/node_modules/rollup/dist/rollup.js:131:5)
    at doNTCallback0 (node.js:430:9)
    at process._tickCallback (node.js:359:13)
```

**Comment by guybedford** [comment]
Sorry, previous comment was incorrect, ignore!

**Comment by guybedford** [comment]
Amazing, well the use case now works perfectly for me here!!

The externals list could be checked against the return value of `resolveId` to avoid the need to return false, but that is really minor.

**Comment by guybedford** [comment]
And I spoke too soon again.... it is still important to have the `resolveId` output check against externals as that is useful to ensure can be normalized into the build.

**Comment by Rich-Harris** [comment]
@guybedford I'm afraid you've lost me! Could you run the issue by me again? Sorry, I'm probably being a bit thick

**Comment by guybedford** [comment]
@Rich-Harris that is certainly understandable given my confounding trail of false hope :) I've deleted all the previous comments, and updated the original description to fully reflect the issue. The original thrown error is no longer provided in the latest 0.25 release either. Let me know if that clarifies?

**Comment by Victorystick** [comment]
@guybedford Do you expect the generated code to point to the _resolved_ external id?

```js
// You want this
import $ from 'npm:jquery@2.1.4/dist/jquery.js';

// rather than that?
import $ from 'jquery';
```

**Comment by guybedford** [comment]
Yes exactly, to be able to control the normalization of the external. Although I understand within the context of node-resolution this may not be desirable, one can provide an externals list based on the resolved externals desired in the output file.

**Comment by guybedford** [comment]
This seems to be fixed in 0.26.
